tempesta_fw/http_sess.h

```diff
@@ -0,0 +1,64 @@
+/*
+ *		Tempesta FW
+ *
+ * Copyright (C) 2017 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+#ifndef __TFW_HTTP_SESS_H__
+#define __TFW_HTTP_SESS_H__
+
+#include "http.h"
+
+/**
+ * Connection, servicing HTTP session.
+ *
+ * @srv_conn	- last used connection;
+ * @main_sg	- primary server group to schedule connection on failovering;
+ * @backup_sg	- backup server group;
+ * @lock	- protects whole @TfwStickyConn;
+ */
+typedef struct {
+	TfwSrvConn		*srv_conn;
+	TfwSrvGroup		*main_sg;
+	TfwSrvGroup		*backup_sg;
+	rwlock_t		lock;
+} TfwStickyConn;
+
+/**
+ * HTTP session descriptor.
+ *
+ * @hmac	- crypto hash from values of an HTTP request;
+ * @hentry	- hash list entry for all sessions hash;
+ * @users	- the session use counter;
+ * @ts		- timestamp for the client's session;
+ * @expire	- expiration time for the session;
+ * @st_conn	- upstream server connection servicing the session;
+ */
+struct tfw_http_sess_t {
+	unsigned char		hmac[SHA1_DIGEST_SIZE];
+	struct hlist_node	hentry;
+	atomic_t		users;
+	unsigned long		ts;
+	unsigned long		expires;
+	TfwStickyConn		st_conn;
+};
+
```

Comment:
In last PR i was advised by @keshonok  to move all the session related definitions from `http.h` to separate header. But  `TfwHttpSess` is required for `TfwHttpReq` in `http.h` and most functions in this header require definitions from `http.h`, so circular dependency take place here.

As for me, advice was very valuable: http.h is too big and placing there more and more definitions only breaks it to loosely connected parts.